src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -74,12 +75,19 @@ protected void onStreamClosed() {
 
         @Override
         public GetSecretValueResponse handleRequest(GetSecretValueRequest request) {
+            return null;
+        }
+
+        @Override
+        public CompletableFuture<GetSecretValueResponse> handleRequestAsync(GetSecretValueRequest request) {
             logger.atDebug("ipc-get-secret-request").log();
+            CompletableFuture<GetSecretValueResponse> responseFut = new CompletableFuture<>();
             try {
                 validateSecretIdAndDoAuthorization(GET_SECRET_VALUE, serviceName, request.getSecretId());
                 logger.atInfo().event("secret-access").kv("Principal", serviceName).kv("secret", request.getSecretId())
                         .log("requested secret");
-                return secretManager.getSecret(request);
+                responseFut.complete(secretManager.getSecret(request));
```

Comment:
this is still synchronous, would need to something like` CompletableFuture.supplyAsync(() -> secretManager.getSecret(request), someExecutorService)`